services/security/authentication/casigned/casigned.go

```diff
@@ -37,6 +44,47 @@ func (c *client) Login(ctx context.Context, group string, identity *security.Ide
 	return &response.Token, nil
 }
 
+func RenewRoutine(group, server string) {
+	renewalAttempt := 0
+	// Waiting for a few seconds to avoid spamming short-lived sdk user
+	time.Sleep(time.Second * 5)
+	for {
+		wssdConfig := auth.WssdConfig{}
+		err := marshal.FromJSONFile(auth.GetWssdConfigLocation(), &wssdConfig)
+		if err != nil {
+			fmt.Printf("Failed to open config file in location %s: %v\n", auth.GetWssdConfigLocation(), err)
+			panic("Failed to open config file")
+		}
+
+		sleepTime, err := renewTime(wssdConfig.ClientCertificate)
+		if err != nil {
+			fmt.Printf("Failed to find sleep time for cert %v \n", err)
+			panic("Failed to find renew time for certificate")
+		}
+		fmt.Printf("Waiting for %v to renew cert\n", sleepTime)
+		time.Sleep(sleepTime)
+		fmt.Printf("Attempting to renew certificate\n")
+		err = auth.RenewCertificates(server, auth.GetWssdConfigLocation())
+		if err != nil {
+			if renewalAttempt > 5 {
+				fmt.Printf("Failed to renew cert to %v \n", err)
+				panic("Failed to renew certificate")
+			}
+			if errors.IsExpired(err) {
+				fmt.Printf("Certificate Expired %v \n", err)
+				panic("Certificate has expired")
+			}
+			renewalAttempt += 1
+			// Wait for 30 seconds before next renewal attempt
+			time.Sleep(time.Second * 30)
```

Comment:
Doesn't this time need to be based of the sleepTime?
And why cap it at 5?  Why not keep letting it run (with a delay in between each attempt) till it expires?

Reply:
It would be awesome if you could display a bomb timer in the logs and play some terrifying music. 
Seriously, the time remaining before expiry might be nice.
Or something similar (the current time, and the time it expires)
Whatever is easier